TextToSpeech/impl/TTSManager.cpp

```diff
@@ -0,0 +1,356 @@
+        std::map<std::string, std::string> configSet;
+        std::regex re("\\s*([a-zA-Z0-9_-]+)\\s*=\\s*([^ ]+).*$");
+
+        while(1) {
+            if(std::getline(configFile, line)) {
+                if(!line.empty() && std::regex_match(line.c_str(), m, re) && m.size() >= 3)
+                    configSet[m[1].str()] = m[2].str();
+            } else
+                break;
+        }
+
+        std::map<std::string, std::string>::iterator it;
+        if((it = configSet.find("TTSEndPoint")) != configSet.end()) {
+            m_defaultConfiguration.setEndPoint(it->second.c_str());
+            configSet.erase(it);
+        }
+
+        if((it = configSet.find("SecureTTSEndPoint")) != configSet.end()) {
+            m_defaultConfiguration.setSecureEndPoint(it->second.c_str());
+            configSet.erase(it);
+        }
+
+        if((it = configSet.find("Language")) != configSet.end()) {
+            m_defaultConfiguration.setLanguage(it->second.c_str());
+            configSet.erase(it);
+        }
+
+        if((it = configSet.find("Voice")) != configSet.end()) {
+            m_defaultConfiguration.setVoice(it->second.c_str());
+            configSet.erase(it);
+        }
+
+        if((it = configSet.find("Volume")) != configSet.end()) {
+            m_defaultConfiguration.setVolume(std::stod(it->second));
+            configSet.erase(it);
+        }
+
+        if((it = configSet.find("Rate")) != configSet.end()) {
+            m_defaultConfiguration.setRate(std::stoi(it->second));
+            configSet.erase(it);
+        }
+
+        m_defaultConfiguration.m_others = std::move(configSet);
+
+        configFile.close();
+    } else {
+        TTSLOG_ERROR("Configuration file \"%s\" is not found, using defaults", configFileName.c_str());
+    }
+
+    TTSLOG_WARNING("TTSEndPoint : %s", m_defaultConfiguration.endPoint().c_str());
+    TTSLOG_WARNING("SecureTTSEndPoint : %s", m_defaultConfiguration.secureEndPoint().c_str());
+    TTSLOG_WARNING("Language : %s", m_defaultConfiguration.language().c_str());
+    TTSLOG_WARNING("Voice : %s", m_defaultConfiguration.voice().c_str());
+    TTSLOG_WARNING("Volume : %lf", m_defaultConfiguration.volume());
+    TTSLOG_WARNING("Rate : %u", m_defaultConfiguration.rate());
+
+    auto it = m_defaultConfiguration.m_others.begin();
+    while( it != m_defaultConfiguration.m_others.end()) {
+        TTSLOG_WARNING("%s : %s", it->first.c_str(), it->second.c_str());
+        ++it;
+    }
+}
+
+TTSManager* TTSManager::create(TTSEventCallback *eventCallback)
+{
+    TTSLOG_TRACE("TTSManager::create");
+    static std::mutex mutex;
+    std::lock_guard<std::mutex> lock(mutex);
+
+    return new TTSManager(eventCallback);
+}
+
+TTSManager::TTSManager(TTSEventCallback *callback) :
+    m_callback(callback),
+    m_speaker(NULL),
+    m_ttsEnabled(false) {
+
+    TTSLOG_TRACE("TTSManager::TTSManager");
+
+    // Load configuration from file
+    loadConfigurationsFromFile(TTS_CONFIGURATION_FILE);
```

Comment:
This should not be allowed in PROD builds.

Reply:
Hi Eugene, could you let us know the concern?

Reply:
This is the general rule. in nutshell, any dev configuration left on the box should not have any impact on prod image

Reply:
We are planning to add the default configuration as a config setting in the ticket RDK-20988.